internal/webhooks/patch_validation_test.go

```diff
@@ -0,0 +1,1372 @@
+									},
+									JSONPatches: []clusterv1.JSONPatch{
+										{
+											Op:   "add",
+											Path: "/spec/template/spec/variableSetting/variableValue1",
+											ValueFrom: &clusterv1.JSONPatchValue{
+												Variable: pointer.String("variableName1"),
+											},
+										},
+									},
+								},
+							},
+						},
+						{
+							Name: "patch2",
+							Definitions: []clusterv1.PatchDefinition{
+								{
+									Selector: clusterv1.PatchSelector{
+										APIVersion: "controlplane.cluster.x-k8s.io/v1beta1",
+										Kind:       "ControlPlaneTemplate",
+										MatchResources: clusterv1.PatchSelectorMatch{
+											ControlPlane: pointer.Bool(true),
+										},
+									},
+									JSONPatches: []clusterv1.JSONPatch{
+										{
+											Op:   "add",
+											Path: "/spec/template/spec/variableSetting/variableValue2",
+											ValueFrom: &clusterv1.JSONPatchValue{
+												Variable: pointer.String("variableName2"),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					Variables: []clusterv1.ClusterClassVariable{
+						{
+							Name:     "variableName1",
+							Required: true,
+							Schema: clusterv1.VariableSchema{
+								OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+									Type: "string",
+								},
+							},
+						},
+						{
+							Name:     "variableName2",
+							Required: true,
+							Schema: clusterv1.VariableSchema{
+								OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+									Type: "string",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "pass if patches uses a name starting with \"builtin\" with no additional variable defined",
+			clusterClass: clusterv1.ClusterClass{
+				Spec: clusterv1.ClusterClassSpec{
+					ControlPlane: clusterv1.ControlPlaneClass{
+						LocalObjectTemplate: clusterv1.LocalObjectTemplate{
+							Ref: &corev1.ObjectReference{
+								APIVersion: "controlplane.cluster.x-k8s.io/v1beta1",
+								Kind:       "ControlPlaneTemplate",
+							},
+						},
+					},
+					Patches: []clusterv1.ClusterClassPatch{
+						{
+							Name: "patch1",
+							Definitions: []clusterv1.PatchDefinition{
+								{
+									Selector: clusterv1.PatchSelector{
+										APIVersion: "controlplane.cluster.x-k8s.io/v1beta1",
+										Kind:       "ControlPlaneTemplate",
+										MatchResources: clusterv1.PatchSelectorMatch{
+											ControlPlane: pointer.Bool(true),
+										},
+									},
+									JSONPatches: []clusterv1.JSONPatch{
+										{
+											Op:   "add",
+											Path: "/spec/template/spec/variableSetting/variableValue2",
+											ValueFrom: &clusterv1.JSONPatchValue{
+												// We only validate variables to see if they start with builtin.
+												Variable: pointer.String("builtin.cluster.namespace"),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
```

Comment:
WDYT about adding comments between the blocks of table tests to make it easier to figure out which validation we're testing, e.g.
```go
// Patch name validation 
or
// validatePatchName
```

```go
// validateJSONPatches
```
etc. further down for the other test "blocks"

I think it would be easier to keep the table tests structured over time and to help the reader know where to look for specific cases.

Reply:
Usually that's not an issue as we have more smaller test funcs for the individual prod funcs. I think it's fine to have one big table test here, it would just help if we have some comments to help navigate those test cases.